src/ovs/aca_ovs_l2_programmer.cpp

```diff
@@ -154,13 +164,28 @@ int ACA_OVS_L2_Programmer::setup_ovs_bridges_if_need()
 
     execute_openflow_command("add-flow br-tun \"table=0,priority=25,in_port=\"vxlan-generic\" actions=resubmit(,4)\"",
                              not_care_culminative_time, overall_rc);
-  } else {
-    // case 3: only one of the br-int or br-tun is there,
-    // Invalid environment so return an error
-    ACA_LOG_CRIT("Invalid environment br-int=%d and br-tun=%d, cannot proceed\n",
-                 br_int_existed, br_tun_existed);
-    overall_rc = EXIT_FAILURE;
+    setup_ovs_bridges_mutex.unlock();
+    // -----critical section ends-----
+
+    ACA_LOG_DEBUG("ACA_OVS_L2_Programmer::setup_ovs_bridges_if_need <--- Exiting, overall_rc = %d\n",
+                  overall_rc);
+
+    return overall_rc;
+  }
+
+  // case 3: only one of the br-int or br-tun is there,
+  // Invalid environment so return an error
+  if (br_int_existed) {
+    ACA_LOG_DEBUG("You have br-int, but you don't have br-tun, please add br-tun by executing command 'ovs-vsctl add-br br-tun', or delete the existing br-int by executing command 'ovs-vsctl del-br br-int' and try again.\n",
+                  br_tun_existed);
+  }
```

Comment:
Hmm, a second thought: in the new implementation, Case I and Case III is not exclusive. It seems that we have to figure out the mystery else-if issue :-)

Reply:
If Case I was hit, it will be returned at line 116, so it shouldn't be a problem.

Reply:
@zzxgzgz Is this PR ready to merge? If so, I would go ahead to merge it.

Reply:
yes, please merge it. thanks